requirements/base.txt

```diff
@@ -1,17 +1,16 @@
-apscheduler>=3.5.1,<3.6
-cliff>=2.2.0,<2.3
+apscheduler>=3.5.1,<=3.7.0
 google-auth>=1.4.1,<1.5
 google-api-python-client>=1.7.7,<1.8
-keystoneauth1>=3.13.0,<3.14
-osc-lib>=1.2.0,<1.3
-os-client-config>=1.28.0,<1.29
-oslo.serialization>=2.28.1,<2.29
-oslo.utils>=3.37.1,<3.38
-paramiko>=2.1.6,<2.2
-python-heatclient>=1.6.1,<1.7
-python-keystoneclient>=3.17.0,<3.18
-python-novaclient>=7.1.2,<7.2
-tenacity>=6.2
+keystoneauth1>=3.13.0,<3.18
+osc-lib>=1.2.0,<2
+os-client-config>=1.28.0,<2
+oslo.serialization>=2.28.1,<3
+oslo.utils>=3.37.1,<3.42.1
```

Comment:
Any particular reason why you'd allow 3.42.0 here, but not 3.42.1? Per semver the only difference between .0 and .1 could possibly be that .1 contains a patch fixing a bug that .0 has.

Reply:
Should this be <=3.42.1 then or <3.43?

Reply:
Per `oslo.utils`' [release history](https://pypi.org/project/oslo.utils/#history), if OpenStack Train is our reference point, I'd suggest `<4`.

Reply:
Ack, done.